### ISSUE TITLE
Other Basemaps Connection Error in iOS 11

### ISSUE THREAD

**Issue by egbunse** [StraboSpot/strabo-mobile#437]
@jessicagood @jasonash When trying to add an "other basemap" using Mapbox, I've run into a problem that @panamint55 thinks is iOS 11 related. I'm on "Add a Map" and choose "Mapbox Styles," enter a title, copy the style URL from Mapbox, and confirm the access token is there from the user info then hit the checkmark. A message box pops up that says "Connection Error! Not able to connect to this map. Check your connection parameters."
What we see happening is the Style URL getting shortened. They get copied beginning with "mapbox://styles/username/map##" but when I hit the checkmark it is getting the "mapbox://styles/" removed. 
I've replicated this (originally working on the KGS Mapbox account) with my Mapbox account on the same device. Doug successfully helped me get the KGS basemap loaded on one of the older iPads this morning which makes us think its OS version related.

**Comment by jessicagood** [comment]
@egbunse I upgraded my iPad to iOS 11 and followed your same steps using Mapbox Styles and it connects fine for me and the map displays as it should.

You're right that the style URL is shortened but the app also does this on Android and web. The app only really needs the end part of the URL but it's easier for the user to just copy the whole URL and for the app to grab (and only save) the part it needs.

I did have some trouble just copying the style URL and access token from the Mapbox website in Safari. I'm not sure if that's something wonky with iOS 11 or if that's how it was before but getting the full values copied to the clipboard was a pain. Maybe something got screwy when you were trying to copy?

Any other ideas for how I can recreate your problem? @jasonash Are you able to test on your device sometime and see if you get what @egbunse is describing?

**Comment by jasonash** [comment]
@jessicagood I tried with my iOS device (10.3.3) and it also worked fine. I did however have quite a bit of difficulty copying the url from mapbox. The copy button on their website doesn't seem to work with mobile safari, so I had to repeatedly try and select and copy. The first couple of times, I actually ended up with an incorrectly copied url, and had to try repeatedly to get it right. This of course is a problem with the mapbox website and safari's copy mechanism. The Strabo app is working as expected.

**Comment by egbunse** [comment]
@jessicagood I'll keep thinking of more info for you. I've tried to copy/paste the values into the Add a Map window from the Chrome and Safari apps with the same results. To get around the weird copying I emailed the info from both accounts (the copy/paste in Mapbox works fine on desktop) and copied into the app that way. This allowed me to see the full access token when I copied. This morning I even typed both the Style URL and Access Token in. I haven't been able to get further than the connection error on this iPad.

**Comment by jessicagood** [comment]
What about other iPads? And/or if you log out and log back in on the troublesome iPad? Or reinstall the app? One way I wasn't expecting that caused me to get the connection error that you described was if the letter case in the style URL was wrong. Mapbox will refuse the connection if the case is wrong. This shouldn't be an issue if you are copying though.

**Comment by egbunse** [comment]
@jessicagood Using one of the field camp iPads on Friday I was able to successfully connect to basemaps on both my account and the KGS Mapbox account (but I think we were on a different Strabo account, Doug gave me the iPad to try). This is why we suspected it was an OS update issue. I can try logging out and/or reinstalling the app. We just downloaded it last week on TestFlight.

**Comment by jessicagood** [comment]
@egbunse Sorry I haven't been able to fix this. I even removed my development version and installed the TestFlight version you're using on my iPad with **iOS 11** and it works fine for me to connect to a Mapbox Style. @jasonash you tried on iOS 10.3.3 but what about 11? Or @panamint55 can you duplicate on your device?

**Comment by egbunse** [comment]
@jessicagood I'm closing this because I finally got it to work on the online app and the iPad in question. What I did was copy the user access token from Mapbox to the User's profile on the online app and save (which I thought I had done, but I guess something got lost when I emailed the code). Then I logged into the account on the iPad and added the basemap successfully. Guess this is a case for setting up all the Mapbox stuff on the online app instead of the device since Mapbox has difficulties copying and displaying the whole code to type in manually. Sorry for all the trouble you and @jasonash had to go to!
